chain/chain/src/chain.rs

```diff
@@ -391,6 +396,60 @@ pub fn check_known(
     check_known_store(chain, block_hash)
 }
 
+// Cache with the mappint from CryptoHash (blocks, chunks) to the number milliseconds that it took to process them.
+// Used only for debugging purposes.
+pub static CRYPTO_HASH_TIMER_RESULTS: Lazy<Mutex<LruCache<CryptoHash, u64>>> =
+    Lazy::new(|| Mutex::new(LruCache::new(10000)));
+
+/// Struct to measure computation times related to different CryptoHashes (for example chunk or block computations).
+/// It stores the data in the global LRU cache, which allows it to be read afterwards.
+///
+/// Example use:
+/// ```rust, ignore
+/// {
+///     let _timer = CryptoHashTime(block.hash);
+///     // Do stuff with the block
+///     // timer gets released at the end of this block
+/// }
+/// ```
+pub struct CryptoHashTimer {
+    key: CryptoHash,
+    start: Instant,
+}
+
+impl CryptoHashTimer {
+    pub fn new(key: CryptoHash) -> Self {
+        CryptoHashTimer { key, start: Clock::instant() }
+    }
+    pub fn get_timer_value(key: CryptoHash) -> Option<u64> {
+        CRYPTO_HASH_TIMER_RESULTS.lock().unwrap().get(&key).cloned()
+    }
+}
+
+impl Drop for CryptoHashTimer {
+    fn drop(&mut self) {
+        CRYPTO_HASH_TIMER_RESULTS
+            .lock()
+            .unwrap()
+            .put(self.key, (Clock::instant() - self.start).as_millis() as u64);
+    }
+}
+
+#[test]
+fn test_crypto_hash_timer() {
+    let crypto_hash = CryptoHash::from_str("s3N6V7CNAN2Eg6vfivMVHR4hbMZeh72fTmYbrC6dXBT").unwrap();
+    // Timer should be missing.
+    assert_eq!(CryptoHashTimer::get_timer_value(crypto_hash), None);
+    let mock_clock_guard = near_primitives::time::MockClockGuard::default();
+    mock_clock_guard.add_instant(Instant::now());
+    mock_clock_guard.add_instant(Instant::now() + std::time::Duration::from_secs(1));
+    {
+        let _timer = CryptoHashTimer::new(crypto_hash);
+    }
+    // Timer should show exactly 1 second (1000 ms).
+    assert_eq!(CryptoHashTimer::get_timer_value(crypto_hash), Some(1000));
```

Comment:
Nice that we can actually write such tests now! Hopefully one day we'll replace global `Clock::instant` with explicitly injected

```rust
fn foo(clock: &Clock) {
  let now = clock.instant();
}
```

Reply:
ack